### PR TITLE
[RSDK-3126] Base Service + Tests

### PR DESCRIPTION
**Pull request by njooma**
Introduce tests for the Base resource, along with its service

### CODE REVIEW

File: lib/src/components/base/service.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  BaseService(this._manager);
+
+  Base _fromManager(String name) {
+    try {
+      return _manager.getResource(Base.getResourceName(name));
+    } catch (e) {
+      throw GrpcError.notFound(e.toString());
+    }
+  }
```
Comment (clintpurser) [inline lib/src/components/base/service.dart:20]: I like how you named this, omitting `base`. I will copy in other services.

I've been thinking how we can abstract some of the boiler plate code thats essentially copied between each service, maybe a service `mixin` would be a good move to abstract this out and include there.

File: lib/src/components/servo/service.dart [hunk 2]
```diff
@@ -11,45 +11,45 @@ class ServoService extends ServoServiceBase {
 
   ServoService(this._manager);
 
-  Servo _servoFromManager(String name) {
+  Servo _fromManager(String name) {
```
Comment (clintpurser) [inline lib/src/components/servo/service.dart:14]: oh! you already got servo, nice. 🙏

File: test/unit_test/components/base_test.dart [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+      expect(base.extra, {'foo': 'bar'});
+    });
+  });
+
+  group('Base RPC Tests', () {
+    late ClientChannel channel;
+    late FakeBase base;
+    late BaseService service;
+    late Server server;
+    late String name;
```
Comment (clintpurser) [inline test/unit_test/components/base_test.dart:176]: `nit` same as above
```suggestion
    const String name = 'base';
```

File: lib/src/components/base/service.dart (hunk 1 above)
Comment (njooma) [reply lib/src/components/base/service.dart:20]: Ya same. Tried abstracting it and ran into issues around not having abstract static functions. Renamed to be just `_fromManager` to make it easier to copy and paste until we figure out a way to refactor it lol

Comment (clintpurser) [reply lib/src/components/base/service.dart:20]: SGTM, I'm down for finding the right abstractions after we have things fleshed out.

**Review by clintpurser — APPROVED** [review]
LGTM - a few nits that are super minor.

**Pull request merged**